server/src/main/java/org/opensearch/action/search/SearchUtils.java

```diff
@@ -40,4 +50,52 @@ public static StepListener<BiFunction<String, String, DiscoveryNode>> getConnect
         }
         return lookupListener;
     }
+
+    /**
+     * Delete list of pits, return success if all reader contexts are deleted ( or not found ).
+     */
+    public static void deletePits(
+        Collection<SearchContextIdForNode> contexts,
+        ActionListener<Integer> listener,
+        ClusterState state,
+        SearchTransportService searchTransportService
+    ) {
+        final Set<String> clusters = contexts.stream()
+            .filter(ctx -> Strings.isEmpty(ctx.getClusterAlias()) == false)
+            .map(SearchContextIdForNode::getClusterAlias)
+            .collect(Collectors.toSet());
+        StepListener<BiFunction<String, String, DiscoveryNode>> lookupListener = getConnectionLookupListener(
+            searchTransportService.getRemoteClusterService(),
+            state,
+            clusters
+        );
+        lookupListener.whenComplete(nodeLookup -> {
+            final GroupedActionListener<Boolean> groupedListener = new GroupedActionListener<>(
+                ActionListener.delegateFailure(
+                    listener,
+                    (l, result) -> l.onResponse(Math.toIntExact(result.stream().filter(r -> r).count()))
+                ),
+                contexts.size()
+            );
+
+            for (SearchContextIdForNode contextId : contexts) {
+                final DiscoveryNode node = nodeLookup.apply(contextId.getClusterAlias(), contextId.getNode());
+                if (node == null) {
+                    groupedListener.onFailure(new OpenSearchException("node not found"));
```

Comment:
Sharing my thought here to refine the requirement..

if node is not found is that as good as context not found or are we considering the case where a node with ES process up was kicked out of cluster and hence still has the PIT contexts alive?